### PR TITLE
Integrate temp admin index book

### PR DESCRIPTION
**Pull request by fadyyousry**
- I removed all javascript files (libraries) and replace them with yarn
- I added a new namespace for the book, author and category model
- I added an index (book, author and category) view for the admin layout.
- I removed the collapse button of the sidebar (the black square)
**Note**: firefox has problems with sidebar

### CODE REVIEW

File: app/controllers/manager/books_controller.rb [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      def author_params
+        params.require(:authors).permit(:names)
+      end
+  
+      def category_params
+        params.require(:categories).permit(:names)
+      end
+  end
+  
+
```
Comment (Magdyx) [inline app/controllers/manager/books_controller.rb:68]: remove space

File: app/helpers/book_helper.rb [hunk 2]
```diff
@@ -1,4 +1,8 @@
 module BookHelper
+    def publisher_name publisher
+        publisher.name if publisher.persisted?
+    end
```
Comment (Magdyx) [inline app/helpers/book_helper.rb:4]: I prefer this method in book model

File: app/javascript/admin/datatable_book.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+			return tr.querySelector("#table_hidden_data").innerHTML;
+		};
+
+		// insert the expand/collapse column
+		var th = document.createElement( 'th' );
+		var td = document.createElement( 'td' );
+		td.innerHTML = '<i data-toggle class="fa fa-plus-square-o text-primary h5 m-none" style="cursor: pointer;"></i>';
+		td.className = "text-center";
+
+		$table
```
Comment (Magdyx) [inline app/javascript/admin/datatable_book.js:23]: Can you fix indentation in this file? I think you need to set your ide to autoconvert tab to spaces.

Comment (fadyyousry) [reply app/javascript/admin/datatable_book.js:23]: done

File: app/controllers/manager/books_controller.rb (hunk 1 above)
Comment (fadyyousry) [reply app/controllers/manager/books_controller.rb:68]: done

File: app/helpers/book_helper.rb (hunk 2 above)
Comment (fadyyousry) [reply app/helpers/book_helper.rb]: done

**Comment by Magdyx** [comment]
> 
> 
>     * I removed all javascript files (libraries) and replace them with yarn
> 
>     * I added a new namespace for the book, author and category model
> 
>     * I added an index (book, author and category) view for the admin layout.
> 
>     * I removed the collapse button of the sidebar (the black square)
>       **Note**: firefox has problems with sidebar

I am using firefox. what issues?

**Pull request merged**